contracts/governance/Vesting/VestingRegistryLogic.sol

```diff
@@ -0,0 +1,301 @@
+	 */
+	function getTeamVesting(
+		address _tokenOwner,
+		uint256 _cliff,
+		uint256 _duration,
+		uint256 _vestingCreationType
+	) public view returns (address) {
+		uint256 type_ = uint256(VestingType.TeamVesting);
+		uint256 uid = uint256(keccak256(abi.encodePacked(_tokenOwner, type_, _cliff, _duration, _vestingCreationType)));
+		return vestings[uid].vestingAddress;
+	}
+
+	/**
+	 * @notice Internal function to deploy Vesting/Team Vesting contract
+	 * @param _tokenOwner the owner of the tokens
+	 * @param _cliff the cliff in seconds
+	 * @param _duration the total duration in seconds
+	 * @param _type the type of vesting
+	 * @param _vestingCreationType the type of vesting created(e.g. Origin, Bug Bounty etc.)
+	 */
+	function _getOrCreateVesting(
+		address _tokenOwner,
+		uint256 _cliff,
+		uint256 _duration,
+		uint256 _type,
+		uint256 _vestingCreationType
+	) internal returns (address) {
+		address vesting;
+		uint256 uid = uint256(keccak256(abi.encodePacked(_tokenOwner, _type, _cliff, _duration, _vestingCreationType)));
+		if (vestings[uid].vestingAddress == address(0)) {
+			if (_type == 1) {
+				vesting = vestingFactory.deployVesting(SOV, staking, _tokenOwner, _cliff, _duration, feeSharingProxy, _tokenOwner);
+			} else {
+				vesting = vestingFactory.deployTeamVesting(SOV, staking, _tokenOwner, _cliff, _duration, feeSharingProxy, vestingOwner);
+			}
+			vestings[uid] = Vesting(_type, _vestingCreationType, vesting);
+			vestingsOf[_tokenOwner].push(uid);
+			isVesting[vesting] = true;
+		}
+		return vestings[uid].vestingAddress;
+	}
+
+	/**
+	 * @notice stores the addresses of Vesting contracts from all three previous versions of Vesting Registry
+	 */
+	function _getDeployedVestings(address _tokenOwner, uint256 _vestingCreationType) internal {
+		uint256 vestingType = 1;
+		uint256 teamVestingType = 0;
+		uint256 uid;
+		uint256 length = vestingRegistries.length;
+		for (uint256 i = 0; i < length; i++) {
+			address vestingAddress = vestingRegistries[i].getVesting(_tokenOwner);
+			if (vestingAddress != address(0)) {
+				VestingLogic vesting = VestingLogic(vestingAddress);
+				uid = uint256(
+					keccak256(abi.encodePacked(_tokenOwner, vestingType, vesting.cliff(), vesting.duration(), _vestingCreationType))
+				);
+				vestings[uid] = Vesting(vestingType, _vestingCreationType, vestingAddress);
+				vestingsOf[_tokenOwner].push(uid);
+				isVesting[vestingAddress] = true;
+			}
+			address teamVestingAddress = vestingRegistries[i].getTeamVesting(_tokenOwner);
+			if (teamVestingAddress != address(0)) {
+				VestingLogic vesting = VestingLogic(teamVestingAddress);
+				uid = uint256(
+					keccak256(abi.encodePacked(_tokenOwner, teamVestingType, vesting.cliff(), vesting.duration(), _vestingCreationType))
+				);
+				vestings[uid] = Vesting(teamVestingType, _vestingCreationType, teamVestingAddress);
+				vestingsOf[_tokenOwner].push(uid);
```

Comment:
This will be used only while migrating previous registries(run only once via scripts) and hence won't create duplicates. The `_vestingCreationType` will be unique for each registry.

Reply:
seeing this only now and it's already a bit late to make  comment, but it is always better to be safe on the contract and do not rely on the function to be used as intended (e.g. only once).